homeassistant/components/landisgyr_heat_meter/const.py

```diff
@@ -50,7 +49,8 @@
         key="heat_previous_year",
         icon="mdi:fire",
         name="Heat usage previous year",
-        native_unit_of_measurement=ENERGY_MEGA_WATT_HOUR,
+        native_unit_of_measurement=UnitOfEnergy.MEGA_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
```

Comment:
Even though it's a diagnostic sensor, I think it should still have the device class set.

Reply:
True. But please have a look at this PR https://github.com/home-assistant/core/pull/80346. This has been open since oktober 14 and resolves this and has many other improvements.

Reply:
#80346 has conflicts, and doesn't use the new unit enumerators.
This PR is part of unlocking #83384.

I suggest to merge this PR, and then rebase #80346 accordingly.